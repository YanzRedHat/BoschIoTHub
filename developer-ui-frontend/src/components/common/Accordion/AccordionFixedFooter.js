/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */
import React from "react";
import ReactDOM from "react-dom";
import styled from "styled-components";
import { withAccordionContext } from "./Accordion";

const AccordionFixedFooterStyled = styled.div`
  &:not(.adding-mode) {
    cursor: pointer;
  }
  position: relative;
  z-index: 1;
  display: flex;
  height: 4rem;
  align-content: center;
  background: #fff;
  border: 1px solid #ccc;
  font-weight: 500;
  padding: 0 1rem;
  box-sizing: border-box;
  color: #757575;
  align-items: center;
  justify-content: flex-start;

  /*
  This could be used to indicate that one can scroll further
  &:before {
    content: "";
    position: absolute;
    top: -100%;
    width: 100%;
    height: 100%;
    background: radial-gradient(
      ellipse at bottom,
      rgba(0, 0, 0, 1) 0%,
      rgba(0, 0, 0, 0) 75%,
      rgba(0, 0, 0, 0) 100%
    );
  } */
`;
/* eslint-disable react/prop-types */
const AccordionFixedFooter = ({ snapFooter, ...props }) =>
  snapFooter ? (
    ReactDOM.createPortal(
      <AccordionFixedFooterStyled snapFooter={snapFooter} {...props} />,
      document.getElementById("fixed-acc-footer-modal")
    )
  ) : (
    <AccordionFixedFooterStyled snapFooter={snapFooter} {...props} />
  );

export default withAccordionContext(AccordionFixedFooter);
