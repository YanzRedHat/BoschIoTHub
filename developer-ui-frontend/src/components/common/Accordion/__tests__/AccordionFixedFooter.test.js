/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */
import React from "react";
import { shallow } from "enzyme";
import AccordionFixedFooter from "components/common/Accordion/AccordionFixedFooter.js";

const foo = i => j => i * j;
const foo2 = foo(2);
const bar = () => ({ baz: foo2, boz: 1 });

describe("AccordionFixedFooter", () => {
  it("should be defined", () => {
    expect(AccordionFixedFooter).toBeDefined();
  });
  it("should render correctly", () => {
    const tree = shallow(<AccordionFixedFooter />);
    expect(tree).toMatchSnapshot();
  });
  it.skip("should have a prop children", () => {
    const obj = bar();
    const tree = shallow(
      <AccordionFixedFooter name="accordionSection test" children={obj} />
    );
    expect(tree.props().children).toEqual(foo2);
  });
});
