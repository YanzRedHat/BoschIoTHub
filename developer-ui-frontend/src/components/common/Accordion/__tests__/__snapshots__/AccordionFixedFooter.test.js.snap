// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionFixedFooter should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
